Fields2Cover build breaks other ROS packages build (Eigen error)

For some reason, building Fields2Cover breaks the build of other ROS packages like spatio_temporal_voxel_layer. Something to do with the Eigen library. I'm not sure if the fault is from Fields2Cover or spatio_temporal_voxel_layer


**Steps to reproduce**

build this dockerfile:

```
FROM osrf/ros:humble-desktop-full
SHELL [ "/bin/bash" , "-c" ]

RUN sudo apt update

WORKDIR /root/ros_ws/src

RUN git clone https://github.com/SteveMacenski/spatio_temporal_voxel_layer -b ros2 && rosdep install --from-paths spatio_temporal_voxel_layer --ignore-src -r -y

# Build spatio_temporal_voxel_layer

WORKDIR /root/ros_ws
RUN source /opt/ros/humble/setup.bash \
 && colcon build --packages-up-to spatio_temporal_voxel_layer

# Build Fields2Cover

RUN sudo apt-get update && \
sudo apt-get install -y --no-install-recommends software-properties-common && \
sudo add-apt-repository ppa:ubuntugis/ppa && \
sudo apt-get update && \
sudo apt-get install -y --no-install-recommends build-essential ca-certificates cmake \
     doxygen g++ git libeigen3-dev libgdal-dev libpython3-dev python3 python3-pip \
     python3-matplotlib python3-tk lcov libgtest-dev libtbb-dev swig libgeos-dev \
     gnuplot libtinyxml2-dev nlohmann-json3-dev  && \
python3 -m pip install gcovr

WORKDIR /root/ros_ws/src

RUN git clone https://github.com/Fields2Cover/Fields2Cover.git

RUN cd /root/ros_ws/src/Fields2Cover \
 && mkdir -p build \
 && cd build \
 && cmake -DBUILD_PYTHON=ON .. \
 && make -j$(nproc) \
 && sudo make install

# Rebuild spatio_temporal_voxel_layer and see that it fails
WORKDIR /root/ros_ws
RUN rm -rf install/spatio_temporal_voxel_layer build/spatio_temporal_voxel_layer
RUN source /opt/ros/humble/setup.bash \
  && colcon build --packages-select spatio_temporal_voxel_layer
```

The error is:

```
 => ERROR [13/13] RUN source /opt/ros/humble/setup.bash   && colcon build --packages-select spatio_temporal_voxel_layer                                                                                                                  5.0s
------                                                                                                                                                                                                                                        
 > [13/13] RUN source /opt/ros/humble/setup.bash   && colcon build --packages-select spatio_temporal_voxel_layer:                                                                                                                             
1.134 Starting >>> spatio_temporal_voxel_layer                                                                                                                                                                                                
4.892 --- stderr: spatio_temporal_voxel_layer                                                                                                                                                                                                 
4.892 COMPONENT = openvdb                                                                                                                                                                                                                     
4.892 CMake Warning (dev) at /opt/ros/humble/share/ament_cmake_ros/cmake/build_shared_libs.cmake:15 (option):                                                                                                                                 
4.892   Policy CMP0077 is not set: option() honors normal variables.  Run "cmake
4.892   --help-policy CMP0077" for policy details.  Use the cmake_policy command to
4.892   set the policy and suppress this warning.
4.892 
4.892   For compatibility with older versions of CMake, option is clearing the
4.892   normal variable 'BUILD_SHARED_LIBS'.
4.892 Call Stack (most recent call first):
4.892   /opt/ros/humble/share/ament_cmake_ros/cmake/ament_cmake_ros-extras.cmake:24 (include)
4.892   /opt/ros/humble/share/ament_cmake_ros/cmake/ament_cmake_rosConfig.cmake:41 (include)
4.892   /opt/ros/humble/share/rosidl_typesupport_fastrtps_c/cmake/rosidl_typesupport_fastrtps_c_generate_interfaces.cmake:21 (find_package)
4.892   /opt/ros/humble/share/ament_cmake_core/cmake/core/ament_execute_extensions.cmake:48 (include)
4.892   /opt/ros/humble/share/rosidl_cmake/cmake/rosidl_generate_interfaces.cmake:286 (ament_execute_extensions)
4.892   CMakeLists.txt:81 (rosidl_generate_interfaces)
4.892 This warning is for project developers.  Use -Wno-dev to suppress it.
4.892 
4.892 cc: warning: EIGEN_MPL2_ONLY: linker input file unused because linking not done
4.892 cc: error: EIGEN_MPL2_ONLY: linker input file not found: No such file or directory
4.892 gmake[2]: *** [CMakeFiles/spatio_temporal_voxel_layer__rosidl_generator_c.dir/build.make:137: CMakeFiles/spatio_temporal_voxel_layer__rosidl_generator_c.dir/rosidl_generator_c/spatio_temporal_voxel_layer/srv/detail/save_grid__functions.c.o] Error 1
4.892 gmake[2]: *** Deleting file 'CMakeFiles/spatio_temporal_voxel_layer__rosidl_generator_c.dir/rosidl_generator_c/spatio_temporal_voxel_layer/srv/detail/save_grid__functions.c.o'
4.892 gmake[1]: *** [CMakeFiles/Makefile2:215: CMakeFiles/spatio_temporal_voxel_layer__rosidl_generator_c.dir/all] Error 2
4.892 gmake[1]: *** Waiting for unfinished jobs....
4.892 c++: warning: EIGEN_MPL2_ONLY: linker input file unused because linking not done
4.892 c++: error: EIGEN_MPL2_ONLY: linker input file not found: No such file or directory
4.892 gmake[2]: *** [CMakeFiles/spatio_temporal_voxel_layer__rosidl_typesupport_cpp.dir/build.make:118: CMakeFiles/spatio_temporal_voxel_layer__rosidl_typesupport_cpp.dir/rosidl_typesupport_cpp/spatio_temporal_voxel_layer/srv/save_grid__type_support.cpp.o] Error 1
4.892 gmake[2]: *** Deleting file 'CMakeFiles/spatio_temporal_voxel_layer__rosidl_typesupport_cpp.dir/rosidl_typesupport_cpp/spatio_temporal_voxel_layer/srv/save_grid__type_support.cpp.o'
4.892 gmake[1]: *** [CMakeFiles/Makefile2:397: CMakeFiles/spatio_temporal_voxel_layer__rosidl_typesupport_cpp.dir/all] Error 2
4.892 c++: warning: EIGEN_MPL2_ONLY: linker input file unused because linking not done
4.892 c++: error: EIGEN_MPL2_ONLY: linker input file not found: No such file or directory
4.892 gmake[2]: *** [CMakeFiles/spatio_temporal_voxel_layer__rosidl_typesupport_introspection_cpp.dir/build.make:123: CMakeFiles/spatio_temporal_voxel_layer__rosidl_typesupport_introspection_cpp.dir/rosidl_typesupport_introspection_cpp/spatio_temporal_voxel_layer/srv/detail/save_grid__type_support.cpp.o] Error 1
4.892 gmake[2]: *** Deleting file 'CMakeFiles/spatio_temporal_voxel_layer__rosidl_typesupport_introspection_cpp.dir/rosidl_typesupport_introspection_cpp/spatio_temporal_voxel_layer/srv/detail/save_grid__type_support.cpp.o'
4.892 gmake[1]: *** [CMakeFiles/Makefile2:371: CMakeFiles/spatio_temporal_voxel_layer__rosidl_typesupport_introspection_cpp.dir/all] Error 2
4.892 c++: warning: EIGEN_MPL2_ONLY: linker input file unused because linking not done
4.892 c++: error: EIGEN_MPL2_ONLY: linker input file not found: No such file or directory
4.892 gmake[2]: *** [CMakeFiles/spatio_temporal_voxel_layer__rosidl_typesupport_fastrtps_cpp.dir/build.make:123: CMakeFiles/spatio_temporal_voxel_layer__rosidl_typesupport_fastrtps_cpp.dir/rosidl_typesupport_fastrtps_cpp/spatio_temporal_voxel_layer/srv/detail/dds_fastrtps/save_grid__type_support.cpp.o] Error 1
4.892 gmake[2]: *** Deleting file 'CMakeFiles/spatio_temporal_voxel_layer__rosidl_typesupport_fastrtps_cpp.dir/rosidl_typesupport_fastrtps_cpp/spatio_temporal_voxel_layer/srv/detail/dds_fastrtps/save_grid__type_support.cpp.o'
4.892 gmake[1]: *** [CMakeFiles/Makefile2:293: CMakeFiles/spatio_temporal_voxel_layer__rosidl_typesupport_fastrtps_cpp.dir/all] Error 2
4.892 gmake: *** [Makefile:146: all] Error 2
4.892 ---
4.892 Failed   <<< spatio_temporal_voxel_layer [3.76s, exited with code 2]
4.945 
4.945 Summary: 0 packages finished [4.22s]
4.945   1 package failed: spatio_temporal_voxel_layer
4.945   1 package had stderr output: spatio_temporal_voxel_layer
```

--- a/spatio_temporal_voxel_layer/CMakeLists.txt
+++ b/spatio_temporal_voxel_layer/CMakeLists.txt
@@ -75,8 +75,6 @@ set(dependencies
 
 set(library_name spatio_temporal_voxel_layer_core)
 
-add_definitions(${EIGEN3_DEFINITIONS})
-
 set(srv_files "srv/SaveGrid.srv")
 rosidl_generate_interfaces(${PROJECT_NAME}
   ${srv_files}

